### PR TITLE
[JENKINS-48766] - Expose information about Remoting versions via API

### PR DESCRIPTION
**Pull request by oleg-nenashev**
See [JENKINS-48766](https://issues.jenkins-ci.org/browse/JENKINS-48766).

- [x] - Unrestrict API introduced in 2.101
- [x] - TcpAgent Listener returns information about minimum Remoting version. Handling on the Remoting side - TODO
- [x] - Jenkins master prints warnings when an agent older than a minimum version connects to the master
- [x] - Add Info about Minimum and Bundled versions to the manifest

### Proposed changelog entries

* RFE: Jenkins defines a minimum Remoting version and prints warnings when an older version is connected

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@reviewbybees @daniel-beck @Vlatombe

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
Also CC @batmat since this API may be useful for extra checks in Version Column Plugin

**Review by ghost — APPROVED** [review]

File: core/src/main/java/hudson/slaves/SlaveComputer.java [hunk 1]
```diff
@@ -545,6 +547,12 @@ public void onClosed(Channel c, IOException cause) {
 
         String slaveVersion = channel.call(new SlaveVersion());
         log.println("Remoting version: " + slaveVersion);
+        VersionNumber agentVersion = new VersionNumber(slaveVersion);
```
Comment (daniel-beck) [inline core/src/main/java/hudson/slaves/SlaveComputer.java]: Not sure whether we still care, but I doubt the fallback `< 1.335` will make a useful version number.

Comment (oleg-nenashev) [reply core/src/main/java/hudson/slaves/SlaveComputer.java]: I don't

**Comment by oleg-nenashev** [comment]
@reviewbybees done

**Pull request merged**